File: python/app/application/user_dto.py
from datetime import datetime

from pydantic import BaseModel, computed_field

from app.utility import json_utility


class UserDto(BaseModel):
    """用户DTO"""

    id: str
    username: str
    # password: str
    nickname: str
    mobile: str
    is_admin: bool
    created_at: datetime

    # pydantic自动序列化
    @computed_field
    def is_not_admin(self) -> bool:
        return not self.is_admin

    model_config = {
        "alias_generator": json_utility.to_camel,
        "populate_by_name": True
    }

    # 手工序列化
    # def to_json(self):
    #     return {
    #         "id": self.id,
    #         "username": self.username,
    #         "nickname": self.nickname,
    #         "mobile": self.mobile,
    #         "isAdmin": self.is_admin,
    #         # "createdAt": self.created_at.isoformat(),
    #         "createdAt": self.created_at,
    #     }

File: dotnet/src/application/UserDto.cs
namespace DddExample.Application
{
    /// <summary>
    /// 用户 DTO
    /// </summary>
    public class UserDto
    {
        public string Id { get; private set; }

        public string Username { get; private set; }

        public string Nickname { get; private set; }

        public string Mobile { get; private set; }

        public bool IsAdmin { get; private set; }

        public DateTime CreatedAt { get; private set; }

        public UserDto(string id, string username, string nickname, string mobile, bool isAdmin, DateTime createdAt)
        {
            Id = id;
            Username = username;
            Nickname = nickname;
            Mobile = mobile;
            IsAdmin = isAdmin;
            CreatedAt = createdAt;
        }
    }
}
